app/controllers/admin/divisions_controller.rb

```diff
@@ -51,8 +51,11 @@ def new
 
   def update
     authorize @division
-
-    if @division.update(division_params)
+    division_params = prep_division_params
+    qb_department_id = division_params.delete(:qb_department_id)
+    @division.update(division_params)
```

Comment:
I think all this logic related to manipulating and fetching the department id should go into `prep_division_params`